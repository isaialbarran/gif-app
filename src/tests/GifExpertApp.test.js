import React from 'react';
import { GifExpertApp } from '../GifExpertApp';
const { shallow } = require('enzyme');

describe('Test component <GifExpertApp>', () => {
  test('Should match with our snapShot', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a list of categories', () => {
    const categories = ['Goku', 'Shinchan'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
