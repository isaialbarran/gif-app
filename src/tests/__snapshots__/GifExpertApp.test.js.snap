// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp> Should match with our snapShot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Test component <GifExpertApp> Should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Shinchan"
      key="Shinchan"
    />
  </ol>
</Fragment>
`;
